logging/logging.go

```diff
@@ -873,33 +873,25 @@ func (l *Logger) writeLogEntries(entries []*logpb.LogEntry) {
 // (for example by calling SetFlags or SetPrefix).
 func (l *Logger) StandardLogger(s Severity) *log.Logger { return l.stdLoggers[s] }
 
-var reCloudTraceContext = regexp.MustCompile(`([a-f\d]+)/([a-f\d]+);o=(\d)`)
+var reCloudTraceContext = regexp.MustCompile(`([a-f\d]+)?(/([a-f\d]+))?(;o=(\d))?`)
 
 func deconstructXCloudTraceContext(s string) (traceID, spanID string, traceSampled bool) {
-	// As per the format described at https://cloud.google.com/trace/docs/troubleshooting#force-trace
+	// As per the format described at https://cloud.google.com/trace/docs/setup#force-trace
 	//    "X-Cloud-Trace-Context: TRACE_ID/SPAN_ID;o=TRACE_TRUE"
 	// for example:
 	//    "X-Cloud-Trace-Context: 105445aa7843bc8bf206b120001000/0;o=1"
 	//
 	// We expect:
-	//   * traceID:         "105445aa7843bc8bf206b120001000"
-	//   * spanID:          ""
-	//   * traceSampled:    true
-	matches := reCloudTraceContext.FindAllStringSubmatch(s, -1)
-	if len(matches) != 1 {
-		return
-	}
+	//   * traceID (optional):         "105445aa7843bc8bf206b120001000"
+	//   * spanID (optional):          "0"
+	//   * traceSampled (optional):    1 (defaults to false)
+	matches := reCloudTraceContext.FindStringSubmatch(s)
 
-	sub := matches[0]
-	if len(sub) != 4 {
-		return
-	}
+	traceID, spanID, traceSampled = matches[1], matches[3], (matches[5] == "1")
```

Comment:
Nit, let's use `matches[5] == "1"` but only after ensuring that we have that component.

Reply:
Regx match will always return 6 components even on nil, or very wrongly formatted strings

Reply:
I meant instead of `(matches[5] == "1")` do `matches[5] == "1"`

Reply:
Ahh sure thing